packages/protocol-parser/src/encodeField.ts

```diff
@@ -8,10 +8,13 @@ export function encodeField<TSchemaAbiType extends SchemaAbiType>(
 ): Hex {
   if (isArrayAbiType(fieldType) && Array.isArray(value)) {
     const staticFieldType = arrayAbiTypeToStaticAbiType(fieldType);
-    return encodePacked(
-      value.map(() => staticFieldType),
-      value
-    );
+    // TODO: we can remove conditional once this is fixed: https://github.com/wagmi-dev/viem/pull/1147
```

Comment:
seems like it got merged already!

Reply:
yep! we'll need to bump viem to get this change so gonna leave this here for now